packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -526,23 +526,38 @@ export class DatabaseCluster extends DatabaseClusterNew {
    * Adds the single user rotation of the master password to this cluster.
    */
   public addRotationSingleUser(options: RotationSingleUserOptions = {}): secretsmanager.SecretRotation {
+    if (options.secret) {
+      //TODO: check given secret is not master secret of this!
+      //TODO: check given secret is not added already!
```

Comment:
If you move the `this.node.tryFindChild(id)` test to the new shared `newSecretRotation()` method, I'm pretty sure you'll get this check for free.

Reply:
This is true for single master secret use case, but not for multiple secrets.